### ISSUE TITLE
Create/Find a component to render a string of html in a nicely formatted/colored way

### ISSUE THREAD

**Issue by ryan953** [getsentry/sentry#36686]
### TL/DR

I want a react component where i can input a string, and get a nice rendered html snippet out of the other end.

Input:
```jsx
const myHtml = `<button class="btn-primary"><span class="btn-icon"><img ...snip...`;

return <FormatHtml
  html={myHtml} // or it's the child? no preference whatsoever.
/>; 
```

Renders:

![Image](https://user-images.githubusercontent.com/187460/179052285-5199fb9c-a606-478b-ae40-53b095fd979b.png)

----

### Motivation

My goal is to show snippets like that one above. Snippets are going to be rendered in a table column, so there’ll be a few snippets to render each time.
Features in order of importance:
1. Syntax highlighting and pretty format are the main features to get.
2. There could be larger snippets too, many lines long.
- If the component could expand/collapse based on indentation that would be amazing.
- Otherwise if it can be wrapped in something that does expand/collapse like [ClippedBox](https://storybook.sentry.dev/?path=/story/utilities-hidden-content-clipped-box--default) that works too. 
3. Horizontal scrolling is an issue, the table shouldn't scroll once I add this component into it, instead the formatted content should scroll with the bounds of it's container. This is probably an easy ask, but something to be aware of.
4. We should remember that Sentry.io supports light/dark mode, so we should make sure it's possible to support that here too

**Not all of these things need to be done in the first version! These are the goal, if we get these then the product will function well.**

I want to have examples and try them out by inserting them into the table that PR #36885 created.
